sscanss/app/dialogs/managers.py

```diff
@@ -943,3 +947,153 @@ def moveDetectorsButtonClicked(self):
+
+        :param label_dict: the dictionary containing the property labels
+        :type label_dict: dict
+        """
+        for row, label_tag in label_dict.items():
+            self._add_property_label(label_tag, row=row)
+
+    def _addMeshPropertyValues(self):
+        """Adds the mesh property values to the table"""
+        memory_str = self._getMemoryUsed()
+        self._add_property_value(memory_str, row=0)
+
+        num_indices, = self.parent_model.sample.indices.shape
+        num_faces = num_indices // 3
+        self._add_property_value(num_faces, row=1)
+
+        num_vertices, _ = self.parent_model.sample.vertices.shape
+        self._add_property_value(num_vertices, row=2)
+
+    def _getMemoryUsed(self):
+        """Calculates the memory used by the sample
+
+        :return memory_str: memory consumed in mb
+        :rtype: string
+        """
+        if isinstance(self.parent_model.sample, Mesh):
+            bytes_used = self.parent_model.sample.vertices.nbytes
+        elif isinstance(self.parent_model.sample, Volume):
+            bytes_used = self.parent_model.sample.data.nbytes
+        elif self.parent_model.sample is None:
+            bytes_used = 0
+        memory_str = self._bytesToMb(bytes_used)
+        return memory_str
+
+    def _bytesToMb(self, bytes_used):
+        """Converts the bytes consumed to mb
+
+        :param bytes_used: the bytes used by the sample
+        :type bytes_used: int
+        :return memory: consumed in mb
+        :rtype: string
+        """
+        bytes_to_mb_factor = 1 / (1024**2)
+        memory = bytes_used * bytes_to_mb_factor
+        memory_str = '0' if memory == 0 else f'{memory:.4f}'
+        return memory_str
+
+    def _addVolumePropertyValues(self):
+        """Adds the Volume property values to the table"""
+        memory_str = self._getMemoryUsed()
+        self._add_property_value(memory_str, row=0)
+
+        x, y, z = self.parent_model.sample.data.shape
+        dimension_str = f'{x} x {y} x {z}'
+        self._add_property_value(dimension_str, row=1)
+
+        voxel_size = self.parent_model.sample.voxel_size
+        voxel_size_str = f'[x: {voxel_size[0]},  y: {voxel_size[1]},  z: {voxel_size[2]}]'
+        self._add_property_value(voxel_size_str, row=2)
+
+        min_intensity = np.min(self.parent_model.sample.data)
+        self._add_property_value(min_intensity, row=3)
+
+        max_intensity = np.max(self.parent_model.sample.data)
+        self._add_property_value(max_intensity, row=4)
+
+    def addSamplePropertyValues(self):
+        """Adds the sample property values to the table"""
+        if isinstance(self.parent_model.sample, Mesh):
+            self._addMeshPropertyValues()
+        elif isinstance(self.parent_model.sample, Volume):
+            self._addVolumePropertyValues()
+        elif self.parent_model.sample is None:
+            self._add_property_value('0', row=0)
+
+    def _addItem(self, item, is_label, row):
```

Comment:
Would it not ne simpler to do 

`def _addItem(self, row, label, value)`

What is the benefit of having code to set the label then code to set the value?

Reply:
Refactored code to remove the extra methods